packages/apps/fortune/exchange-oracle/server/src/modules/job/job.service.ts

```diff
@@ -282,4 +282,30 @@ export class JobService {
       throw new NotFoundException('Unable to get manifest');
     } else return manifest;
   }
+
+  public async getAddressesByAssignmentId(assignmentId: number): Promise<{
+    workerAddress: string;
+    escrowAddress: string;
+    chainId: number;
+  }> {
+    const assignment = await this.assignmentRepository.findOne({
+      where: { id: assignmentId },
+      relations: ['job'],
+    });
+    if (!assignment) {
+      throw new NotFoundException('Assignment not found');
+    }
+
+    const { workerAddress, job } = assignment;
+    if (!job) {
+      throw new NotFoundException('Job not found for the assignment');
+    }
+
+    const { escrowAddress, chainId } = job;
+    return {
+      workerAddress,
+      escrowAddress,
+      chainId,
+    };
+  }
```

Comment:
Why have you created this method? It doesn't have any reference apart from the tests. 
Anyway, as you have the `assignmentId` in the `SolveJob` method, you should modify this one to, instead of searching the task by `escrowAddress` and `worker`, use the `assignmentId`

Reply:
Fixed